File: backend/src/main/java/by/fpmibsu/findafriend/controller/commands/animaladverts/CreateAnimalAdvertCommand.java
package by.fpmibsu.findafriend.controller.commands.animaladverts;

import by.fpmibsu.findafriend.application.mediatr.Request;

import java.util.Date;
import java.util.List;

public class CreateAnimalAdvertCommand extends Request<Boolean> {

    public String title;
    public String description;
    public String animalType;
    public int placeId;
    public Date birthdate;
    public int sex;
    public boolean isCastrated;
    public List<String> photos; // base64 strings
}

File: BackendNET/Backend/Commands/AnimalAdverts/CreateAnimalAdvert/CreateAnimalAdvertCommand.cs
using System.Text.Json.Serialization;
using Backend.Entities;
using MediatR;

namespace Backend.Commands.AnimalAdverts.CreateAnimalAdvert;

public class CreateAnimalAdvertCommand : IRequest<bool>
{
    [JsonIgnore] public int UserId { get; set; }
    public string Title { get; set; } = "";
    public string Description { get; set; } = "";
    public string AnimalType { get; set; } = "";
    public int PlaceId { get; set; }
    public DateOnly BirthDate { get; set; }
    public AnimalAdvert.AnimalSex Sex { get; set; }
    public bool IsCastrated { get; set; }
    public List<string> Photos { get; set; } = null!; // base64
}